cpp/src/arrow/engine/substrait/options.h

```diff
@@ -65,8 +65,8 @@ enum class ARROW_ENGINE_EXPORT ConversionStrictness {
   BEST_EFFORT,
 };
 
-using NamedTableProvider =
-    std::function<Result<compute::Declaration>(const std::vector<std::string>&)>;
+using NamedTableProvider = std::function<Result<compute::Declaration>(
+    const std::vector<std::string>&, const std::shared_ptr<Schema>)>;
```

Comment:
Please use `const Schema&`.  `shared_ptr` should only be needed if the caller is likely to extend ownership.  In the rare case that is needed the caller can just make their own copy.

If you do think it is likely that caller will need to extend ownership you can use `const std::shared_ptr<Schema>&` (`const &` and not just `const`).

Reply:
Thanks sense. Will push another revision.

Reply:
Changed to `const Schema&`